hopsworks-IT/src/test/ruby/spec/jwt_spec.rb

```diff
@@ -152,4 +152,62 @@
     end
     
   end
+
+  describe "Elk JWT tokens" do
+
+    after :all do
+      clean_projects
+    end
+
+    context "#users" do
+      before :all do
+        with_valid_session
+        @proj = create_project_by_name("Project_DEMO_#{Time.now.to_i}")
```

Comment:
cleanup the project once you are done. do not leave garbage behind

Reply:
Isn't clean_projects doing that?

Reply:
can you please check this. I think that since you reset_session in your tests, you will not get a correct cleanup.
with_valid_session checks the @cookies and if it is null, it creates a new session:
https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-IT/src/test/ruby/spec/helpers/session_helper.rb#L41
reset_session sets the @cookies to null
https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-IT/src/test/ruby/spec/helpers/session_helper.rb#L131
and the session with user:
and create_session_with_user does not store the cookies, token... in the global variables
https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-IT/src/test/ruby/spec/helpers/session_helper.rb#L140
so after a create_session_with_user I think the @cookies will be null, and thus the cleanup, having no @cookies will create a new user session, that will not have the rights to delete your project
https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-IT/src/test/ruby/spec/helpers/project_helper.rb#L147
@ErmiasG can you confirm/infirm this

Reply:
ok, so I will try to add the fix to the ruby code later to behave as you expect